File: android/src/main/java/org/ligi/passandroid/model/pass/PassBarCodeFormat.java
package org.ligi.passandroid.model.pass;

import com.google.zxing.BarcodeFormat;

public enum PassBarCodeFormat {
    PDF_417,
    AZTEC,
    CODE_39,
    CODE_128,
    QR_CODE,
    EAN_13;

    public boolean isQuadratic() {
        switch (this) {
            case QR_CODE:
            case AZTEC:
                return true;

            default:
                return false;
        }
    }

    public BarcodeFormat getZxingBarCodeFormat() {
        switch (this) {
            case QR_CODE:
                return BarcodeFormat.QR_CODE;
            case AZTEC:
                return BarcodeFormat.AZTEC;
            case CODE_39:
                return BarcodeFormat.CODE_39;
            case CODE_128:
                return BarcodeFormat.CODE_128;
            case PDF_417:
                return BarcodeFormat.PDF_417;
            case EAN_13:
                return BarcodeFormat.EAN_13;

            default:
                return BarcodeFormat.QR_CODE;
        }
    }
}

File: android/src/main/java/org/ligi/passandroid/model/pass/PassBarCodeFormat.kt
package org.ligi.passandroid.model.pass

import com.google.zxing.BarcodeFormat

enum class PassBarCodeFormat {

    AZTEC,
    CODE_39,
    CODE_128,
    DATA_MATRIX,
    EAN_13,
    ITF,
    PDF_417,
    QR_CODE;

    fun isQuadratic() = when (this) {
        QR_CODE, AZTEC -> true
        else -> false
    }

    fun zxingBarCodeFormat() = when (this) {
        AZTEC -> BarcodeFormat.AZTEC
        CODE_39 -> BarcodeFormat.CODE_39
        CODE_128 -> BarcodeFormat.CODE_128
        DATA_MATRIX -> BarcodeFormat.DATA_MATRIX
        EAN_13 -> BarcodeFormat.EAN_13
        ITF -> BarcodeFormat.ITF
        PDF_417 -> BarcodeFormat.PDF_417
        QR_CODE -> BarcodeFormat.QR_CODE

        else -> BarcodeFormat.QR_CODE
    }
}
